config/src/config/node_config.rs

```diff
@@ -286,4 +459,114 @@ mod test {
         let merged_node_config = merge_node_config(node_config, override_node_config);
         assert!(matches!(merged_node_config, Err(Error::Unexpected(_))));
     }
+
+    #[test]
+    fn test_override_node_config_no_diff() {
+        let override_config = OverrideNodeConfig::new(NodeConfig::default(), NodeConfig::default());
+        let diff_yaml = override_config.get_yaml().unwrap();
+        assert_eq!(diff_yaml, serde_yaml::Value::Null);
+    }
+
+    #[test]
+    fn test_override_node_config_with_bool() {
+        let mut override_config =
+            OverrideNodeConfig::new(NodeConfig::default(), NodeConfig::default());
+        let config = override_config.config_mut();
+        config.api.enabled = false;
+
+        let diff_yaml = override_config.get_yaml().unwrap();
+        let expected_yaml: serde_yaml::Value = serde_yaml::from_str(
+            r#"
+                api:
+                    enabled: false
+                "#,
+        )
+        .unwrap();
+        assert_eq!(diff_yaml, expected_yaml);
+    }
+
+    #[test]
+    fn test_override_node_config_with_enum() {
+        let mut override_config =
+            OverrideNodeConfig::new(NodeConfig::default(), NodeConfig::default());
+        let config = override_config.config_mut();
+        config.base.waypoint = WaypointConfig::FromFile(PathBuf::from("test"));
```

Comment:
nit: empty line after this to match the formatting of the other tests? 😄 (I actually found it pretty helpful for readability).